### PR TITLE
Require the "right" sso (OR right email) for self-registration in autologin case.

### PR DESCRIPTION
**Pull request by maparent**
Also repair next-view on autologin.

### CODE REVIEW

File: assembl/models/social_auth.py [hunk 1]
```diff
@@ -135,6 +135,15 @@ class SocialAuthAccount(
     def provider(self):
         return self.identity_provider.provider_type
 
+    @property
+    def provider_with_idp(self):
+        provider = self.provider
+        if provider == 'saml':
+            idp_name = self.uid.split(':')[0]
```
Comment (ayyazdaniaryan) [inline assembl/models/social_auth.py]: A comment on a typical shape `self.uid` would go far here.

File: assembl/models/discussion.py [hunk 2]
```diff
@@ -359,18 +359,28 @@ def get_discussion_urls(self):
 
     def check_authorized_email(self, user):
         # Check if the user has a verified email from a required domain
+        from .social_auth import SocialAuthAccount
         require_email_domain = self.preferences['require_email_domain']
-        if not require_email_domain:
+        autologin_backend = self.preferences['authorization_server_backend']
```
Comment (ayyazdaniaryan) [inline assembl/models/discussion.py]: Is there anywhere in the discussion preferences or in a documentation page where we can describe the shape and content of this pref? If it was an ini thing, there would be a clear trail of what this is supposed to entail, for example...

File: assembl/models/social_auth.py (hunk 1 above)
Comment (ayyazdaniaryan) [inline assembl/models/social_auth.py]: When does this also get used?

Comment (maparent) [inline assembl/models/social_auth.py]: The IDP name is used all over the place, but this new method is only used in Discussion.check_authorized_email

**Pull request merged**